app/controllers/utils/package.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import play.api.Logger
+import play.api.mvc.Results._
+import play.api.mvc.{Request, Result}
+import uk.gov.hmrc.play.frontend.exceptions.ApplicationException
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{CanAwait, ExecutionContext, Future}
+import scala.util.Try
+
+package object utils {
+  implicit class RecoverableFuture(future: Future[Result]) extends Future[Result] {
+
+    override def onComplete[U](f: (Try[Result]) => U)(implicit executor: ExecutionContext): Unit = future.onComplete(f)
+    override def isCompleted: Boolean = future.isCompleted
+    override def value: Option[Try[Result]] = future.value
+    override def ready(atMost: Duration)(implicit permit: CanAwait): RecoverableFuture.this.type = ready(atMost)
+    override def result(atMost: Duration)(implicit permit: CanAwait): Result = future.result(atMost)
+
+    def recoverToStart(homeLink:String, sessionTimeoutUrl: String)(implicit request: Request[_], ec: ExecutionContext): Future[Result] =
+      future.recover {
+        case e: Exception =>
```

Comment:
This could be for any other exception rather than just None.get, should be more explicit. Like GatewayTimeouts an the like.